src/Controller/CRUDController.php

```diff
@@ -1008,15 +1016,15 @@ protected function configure(): void
         }
 
         try {
-            $this->admin = $this->container->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
+            $this->admin = $this->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
         } catch (\InvalidArgumentException $e) {
             throw new \RuntimeException(sprintf(
                 'Unable to find the admin class related to the current controller (%s)',
                 static::class
             ));
         }
 
-        $this->templateRegistry = $this->container->get(sprintf('%s.template_registry', $this->admin->getCode()));
+        $this->templateRegistry = $this->admin->getTemplateRegistry();
         if (!$this->templateRegistry instanceof TemplateRegistryInterface) {
```

Comment:
I think this can't happen
```
public function getTemplateRegistry(): MutableTemplateRegistryInterface;
```

Reply:
I used the `hasTemplateRegistry` method to ensure the get will not throw the logic exception